datafusion/expr/src/logical_plan/dml.rs

```diff
@@ -39,46 +39,29 @@ pub struct CopyTo {
     /// Determines which, if any, columns should be used for hive-style partitioned writes
     pub partition_by: Vec<String>,
     /// Arbitrary options as tuples
-    pub copy_options: CopyOptions,
-}
-
-/// When the logical plan is constructed from SQL, CopyOptions
-/// will contain arbitrary string tuples which must be parsed into
-/// FileTypeWriterOptions. When the logical plan is constructed directly
-/// from rust code (such as via the DataFrame API), FileTypeWriterOptions
-/// can be provided directly, avoiding the run time cost and fallibility of
-/// parsing string based options.
-#[derive(Clone)]
-pub enum CopyOptions {
-    /// Holds StatementOptions parsed from a SQL statement
-    SQLOptions(StatementOptions),
-    /// Holds FileTypeWriterOptions directly provided
-    WriterOptions(Box<FileTypeWriterOptions>),
+    pub format_options: FormatOptions,
+    ///
+    pub source_option_tuples: HashMap<String, String>,
```

Comment:
Does source_option_tuples hold any option which does not fall into a format specific namespace?

Reply:
No, it does not. This may be handled better with `Option`. I will check that.